client/app/bundles/HelloWorld/startup/HelloWorldApp.jsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import createStore from '../store/helloWorldStore';
+import HelloWorld from '../containers/HelloWorld';
+
+// See documentation for https://github.com/rackt/react-redux.
+// This is how you get props from the Rails view into the redux store.
+// This code here binds your smart component to the redux store.
+const HelloWorldApp = props => {
+  const store = createStore(props);
+  const reactComponent = (
+    <Provider store={store}>
+      <HelloWorld />
+    </Provider>
+  );
+  return reactComponent;
+};
+
+export default HelloWorldApp;
```

Comment:
This is our top level component binding the container component "containers/HelloWorld" to your redux store.

Note, we're **NOT** returning a simple React component. We're returning a function that takes `props` and returns a React component.